packages/components/src/components/chat/Chat.vue

```diff
@@ -171,6 +173,7 @@ export default {
       enableScrollLog: false, // Auto-scroll can be tricky, so there is special logging to help debug it.
       codeSelections: [] as CodeSelection[],
       appmaps: [] as string[],
+      isAssistantMessageActive: false,
```

Comment:
There's a `complete` property on each message, indicating whether or not the response has completed. You could use a computed property to determine whether or not there is an active system response instead of tracking this state manually and separately.

```ts
{

  // ...

  computed: {
    isResponseStreaming(): boolean {
      return this.getMessage({ complete: false, isUser: false }) !== undefined;
    }
  }

  // ...

}

Reply:
It may seem nitpicky but UI changes are very hard to fix or revert once they are in the wild.

Reply:
I removed the extra state variable and applied @dustinbyrne's suggestion. Thank you.